src/main/java/com/dynatrace/openkit/protocol/Beacon.java

```diff
@@ -778,6 +833,22 @@ private void addKeyValuePair(StringBuilder builder, String key, String stringVal
         builder.append(encodedValue);
     }
 
+    /**
+     * Serialization helper method for adding key/value pairs with string values
+     *
+     * if the string value turns out to be null the key value pair is not added
+     * to the string builder
+     *
+     * @param builder The string builder storing serialized data.
+     * @param key The key to add.
+     * @param stringValue The value to add.
+     */
+    private void addKeyValuePairIfNotNull(StringBuilder builder, String key, String stringValue) {
+        if(stringValue != null && stringValue.length() > 0) {
```

Comment:
Be careful here please. The method in C# is only checking for `null` strings and forbidding them.
There is no statement about empty strings, therefore empty strings are allowed.

Reply:
Space between `if` and `(`